tools/kagami/README.md

```diff
@@ -136,7 +137,13 @@ You can also choose output format:
  - Generate a synthetic genesis block in JSON format and write the `n` domains, `m` accounts per domain and `p` assets per domain:
 
     ```bash
-    kagami genesis --synthetic --domains n --accounts-per-domain m --assets-per-domain p
+    kagami genesis synthetic --domains n --accounts-per-domain m --assets-per-domain p
+    ```
+
+ - Generate a default genesis block in JSON format and provide path to the validator file (it could be absolute or relative to genesis file)
+
+    ```bash
+    kagami genesis --validator-path ./validator.wasm
```

Comment:
is this argument always required? I would make it positional if so

Reply:
No, you can omit it and genesis will be generated with validator inlined (previous behavior).

Reply:
why would we want that?

Reply:
i thought that it may be more convenient for some setups to provide everything in single file and want to remain backwards compatible

Reply:
I'm not fan of this. We still have a privilege to not be backwards compatible in this case. It's added complexity for no tangible reason. Having said that, I don't mind supporting both `kagami genesis` and `kagami genesis --validator-path ./validator.wasm`. But both commands should produce two files IMO

Reply:
At the moment we don't produce 2 files, we only producing genesis file with provided path. And validator itself should be generated in separate command.

Reply:
> I'm not fan of this. We still have a privilege to not be backwards compatible in this case. It's added complexity for no tangible reason. Having said that, I don't mind supporting both `kagami genesis` and `kagami genesis --validator-path ./validator.wasm`. But both commands should produce two files IMO

In any case i don't mind to provide only new behaviour. I just don't want to cause trouble to sdk devs and devops. Because @appetrosyan was against removal of configuration params, so here i thought in the same vein.

Reply:
I'm aware of your motivation. As you wish. It's good

Reply:
Let's then wait for other reviewer opinion on this matter.

Reply:
> Because @appetrosyan was against removal of configuration params, so here i thought in the same vein.

That is correct. However, thankfully, here nobody's ever worked with the inline validators outside the core team, so we have flexibility. 

Some software has insane requirements. Sometimes they'll limit sizes of files, sometimes subvolumes that can be mounted shall be restricted. Given that we support both `inline` and included validators, I strongly suggest keeping the functionality in. 

From the perspective of UX, the default should be what we prefer for our CI. I'd add `kagami genesis --inline` as an alternative to `kagami genesis` which warns you that it created a file by default and tells you about the final form: `kagami genesis --compiled-validator-path <path>` for the override.

Reply:
then I would make it part of the serialization step. IMO `RawGenesisBlock` should always work with _Inline_ validator, but in `kagami` or during serialization you can convert the file path to inline validator. @Erigara do you think that is possible? That way you would also be truly backwards compatible because with the current approach you have modified schema

Reply:
then I would make it part of the serialization step. IMO RawGenesisBlock should always work with Inline validator, but in kagami, during deserialization or in constructor you can convert the file path to inline validator. @Erigara do you think that is possible? That way you would also be truly backwards compatible because with the current approach you have modified schema

two files are then just a convenience

Reply:
we can consider deserialization approach for configs and other files to split them up into smaller constituents for reasons such as portability or file size

Reply:
>  do you think that is possible?

@mversic, it was exactly approach i took initially, unfortunately as you said we lose ability to generate proper genesis in kagami and that's why i decide to took another approach. Imo manually move validator into separate file would be pain in the ass.

Reply:
fine by me